flypipe/catalog/catalog.py

```diff
@@ -25,34 +25,29 @@ def __init__(self):
 
     def register_node(self, node, successor=None, node_graph=None):
         if isinstance(node, NodeFunction):
-            expanded_nodes = node.expand(None)
-            self.register_node(expanded_nodes[-1], successor, node_graph)
-        else:
-            if node.node_graph is not None:
-                # The node graph gives us certain information about the nodes in the context of a single run, use this
-                # if available.
-                node_graph = node.node_graph
-            if node.key not in self.nodes:
-                self.nodes[node.key] = CatalogNode(node, node_graph)
-            if node.group:
-                if node.group not in self.groups:
-                    self.groups[node.group] = Group(node.group)
-                self.groups[node.group].add_node(node)
-            if successor:
-                self.nodes[node.key].register_successor(successor)
-            for input_node in node.input_nodes:
-                self.register_node(input_node.node, node, node_graph)
+            raise RuntimeError("Can not register node functions")

```

Comment:
TypeError would be more appropriate here I think, as RuntimeError is meant to be used when "error is detected that doesn't fall into any of the other categories" (https://docs.python.org/3/library/exceptions.html#RuntimeError) but no big deal